expression/chunk_executor.go

```diff
@@ -391,3 +444,13 @@ func VectorizedFilterConsiderNull(ctx sessionctx.Context, filters []Expression,
 	}
 	return selected, isNull, nil
 }
+
+// vectorizedFilter filters by vector.
+func vectorizedFilter(ctx sessionctx.Context, filters []Expression, iterator *chunk.Iterator4Chunk, selected []bool, isNull []bool) ([]bool, []bool, error) {
+	selected, isNull, err := VecEvalBool(ctx, filters, iterator.GetChunk(), selected, isNull)
```

Comment:
In keeping with previous implementations. When the function meets the error, it will retrun ```nil, nil, err```(see [here](https://github.com/pingcap/tidb/blob/master/expression/chunk_executor.go#L383) for more details). But I think both approaches are correct. What's your opinion? @XuHuaiyu